hooks/useGovernedMultiTypeAccounts.ts

```diff
@@ -0,0 +1,61 @@
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import { useCallback, useEffect, useState } from 'react'
+import useGovernanceAssets from './useGovernanceAssets'
+
+export default function useGovernedMultiTypeAccounts() {
+  const {
+    getMintWithGovernances,
+    governancesArray,
+    governedTokenAccounts,
+  } = useGovernanceAssets()
+
+  const [governedMultiTypeAccounts, setGovernedMultiTypeAccounts] = useState<
+    GovernedMultiTypeAccount[]
+  >([])
+
+  const getGovernedMultiTypeAccounts = useCallback(async (): Promise<
+    GovernedMultiTypeAccount[]
+  > => {
+    const mintWithGovernances = await getMintWithGovernances()
+
+    return governancesArray.map((gov) => {
+      const governedTokenAccount = governedTokenAccounts.find(
+        (x) => x.governance?.pubkey.toBase58() === gov.pubkey.toBase58()
+      )
+      if (governedTokenAccount) {
+        return governedTokenAccount as GovernedMultiTypeAccount
+      }
+
+      const mintGovernance = mintWithGovernances.find(
+        (x) => x.governance?.pubkey.toBase58() === gov.pubkey.toBase58()
+      )
+      if (mintGovernance) {
+        return mintGovernance as GovernedMultiTypeAccount
+      }
+
+      return {
+        governance: gov,
+      }
+    })
+  }, [JSON.stringify(governedTokenAccounts), JSON.stringify(governancesArray)])
+
+  useEffect(() => {
+    let abort = false
+
+    ;(async () => {
```

Comment:
A lot of people are confused by IIFEs, I would just use `.then()` here tbh

Reply:
`IIFE` vs `Promise.then` idk which one is the more explicit.

but `IIFE` is used already at multiple occasion in the codebase

Reply:
Ok, then yeah nvm